mitype/app.py

```diff
@@ -300,7 +302,9 @@ def update_state(self, win):
 
             win.addstr(" " + self.current_speed_wpm + " ", curses.color_pair(1))
             win.addstr(" WPM ")
-
+            self.time_taken = time.strftime(" X%Mmins X%Ssec ", time.gmtime(self.end_time-self.start_time)).replace('X0','X').replace('X','')
```

Comment:
Let's just keep time in seconds e.g. 67 seconds and not 1 min 7 seconds.
Try to get time more precisely e.g. 67.4 seconds.
Also, this line is very cryptic.
Just find the difference `end_time - start_time` and print it by passing `"Time taken : " + difference` to addstr.